lib/sync-DataSetModel.js

```diff
@@ -349,6 +355,14 @@ var self = {
     syncUtil.doLog(datasetClient.datasetId, 'silly', 'Now there are ' + datasetClient.pendingCallbacks.length + ' calbacks for datasetclient id =' + datasetClient.id, datasetClient);
   },
 
+  asyncGetClientHash: function(queryParams, metaData, cb) {
+    return syncUtil.asyncGenerateHash(queryParams, function(err, queryParamsHash) {
```

Comment:
Probably not. But normally the meta and query objects are very small, it might be no differences in terms of performance.